tests/unittests/CoreGraphics/CGGeometryTests.mm

```diff
@@ -52,4 +52,122 @@
     CGSize inverseTransformedSize = CGSizeApplyAffineTransform(transformedSize, CGAffineTransformInvert(rotate));
     EXPECT_NEAR(size.width, inverseTransformedSize.width, 0.001);
     EXPECT_NEAR(size.height, inverseTransformedSize.height, 0.001);
-}
\ No newline at end of file
+}
+
+class CGGeometryDivRect : public ::testing::TestWithParam<::testing::tuple<CGRect, CGFloat, CGRectEdge, CGRect, CGRect>> {};
+
+TEST_P(CGGeometryDivRect, RectDivide) {
+    CGRect rect = ::testing::get<0>(GetParam());
+    CGFloat amount = ::testing::get<1>(GetParam());
+    CGRectEdge edge = ::testing::get<2>(GetParam());
+    CGRect rSlice = ::testing::get<3>(GetParam());
+    CGRect rRemainder = ::testing::get<4>(GetParam());
+
+    CGRect slice, remainder;
+    CGRectDivide(rect, &slice, &remainder, amount, edge);
+
+    EXPECT_TRUE(CGRectEqualToRect(rSlice, slice));
+    EXPECT_TRUE(CGRectEqualToRect(rRemainder, remainder));
+}
+
+std::tr1::tuple<CGRect, CGFloat, CGRectEdge, /*slice*/ CGRect, /*remainder*/ CGRect> const cCGRectDivideResults[] = {
+
+    ::testing::make_tuple(CGRectMake(0, 0, 100, 100),
+                          10.35,
+                          CGRectMaxXEdge,
+                          CGRectMake(89.65, 0, 10.35, 100),
+                          CGRectMake(0, 0, 89.65, 100)),
+    ::testing::make_tuple(CGRectMake(0, 0, 100, 100),
+                          10.35,
+                          CGRectMinXEdge,
+                          CGRectMake(0, 0, 10.35, 100),
+                          CGRectMake(10.35, 0, 89.65, 100)),
+    ::testing::make_tuple(CGRectMake(0, 0, 100, 100),
+                          10.35,
+                          CGRectMaxYEdge,
+                          CGRectMake(0, 89.65, 100, 10.35),
+                          CGRectMake(0, 0, 100, 89.65)),
+    ::testing::make_tuple(CGRectMake(0, 0, 100, 100),
+                          10.35,
+                          CGRectMinYEdge,
+                          CGRectMake(0, 0, 100, 10.35),
+                          CGRectMake(0, 10.35, 100, 89.65)),
+    ::testing::make_tuple(CGRectMake(0, 0, 900, 800), -50, CGRectMaxXEdge, CGRectMake(900, 0, 0, 800), CGRectMake(0, 0, 900, 800)),
```

Comment:
might be worth forcing these to format the same as the others and disabling clang-format in this zone